endpoint.bal

```diff
@@ -0,0 +1,390 @@
+        string amzTarget = AMAZON_SQS_API_VERSION + FULL_STOP + ACTION_DELETE_MESSAGE;
+        string|error receiptHandleEncoded = url:encode(receiptHandle, UTF_8);
+        if (receiptHandleEncoded is string) {
+            map<string> parameters = {};
+            parameters[PAYLOAD_PARAM_ACTION] = ACTION_DELETE_MESSAGE;
+            parameters[PAYLOAD_PARAM_RECEIPT_HANDLE] = receiptHandleEncoded;
+            http:Request|error request = self.generatePOSTRequest(amzTarget, queueResourcePath, 
+                                         self.buildPayload(parameters));
+            if (request is http:Request) {
+                var httpResponse = self.clientEp->post(queueResourcePath, request);
+                xml|ResponseHandleFailed response = handleResponse(httpResponse);
+                if (response is xml) {
+                    return isXmlDeleteResponse(response);
+                } else {
+                    return error OperationError(OPERATION_ERROR_MSG, response);
+                }
+            } else {
+                return error OperationError(OPERATION_ERROR_MSG, request);
+            }
+        } else {
+            return error OperationError(OPERATION_ERROR_MSG, receiptHandleEncoded);
+        }
+    }
+
+    # Delete queue(s)
+    #
+    # + queueResourcePath - Resource path to the queue from the host address. e.g.: /610968236798/myQueue.fifo
+    # + return - Whether the queue(s) were successfully deleted or whether an error occurred
+    @display {label: "Delete the queue"}
+    remote isolated function deleteQueue(@display {label: "Resource path to queue"} string queueResourcePath)
+                                  returns @tainted @display {label: "Delete status"} boolean|OperationError {
+        string amzTarget = AMAZON_SQS_API_VERSION + FULL_STOP + ACTION_DELETE_QUEUE;
+        map<string> parameters = {};
+        parameters[PAYLOAD_PARAM_ACTION] = ACTION_DELETE_QUEUE;
+        http:Request|error request = self.generatePOSTRequest(amzTarget, queueResourcePath, 
+                                     self.buildPayload(parameters));
+        if (request is http:Request) {
+            var httpResponse = self.clientEp->post(queueResourcePath, request);
+            xml|ResponseHandleFailed response = handleResponse(httpResponse);
+            if (response is xml) {
+                return isXmlDeleteQueueResponse(response);
+            } else {
+                return error OperationError(OPERATION_ERROR_MSG, response);
+            }
+        } else {
+            return error OperationError(OPERATION_ERROR_MSG, request);
+        }
+    }
+
+    private isolated function buildPayload(map<string> parameters) returns string {
+        string payload = EMPTY_STRING;
+        int parameterNumber = 1;
+        foreach var [key, value] in parameters.entries() {
+            if (parameterNumber > 1) {
+                payload = payload + AMBERSAND;
```

Comment:
Wrong wording AMBERSAND must be AMPERSAND

Reply:
Corrected the typo